mkosi/config.py

```diff
@@ -0,0 +1,275 @@
+
+
+def config_parse_script(name: str, dest: str, value: Optional[str], namespace: argparse.Namespace) -> None:
+    if value is not None:
+        if not Path(value).exists():
+            die(f"{value} does not exist")
+        if not os.access(value, os.X_OK):
+            die(f"{value} is not executable")
+
+    config_parse_path(name, dest, value, namespace)
+
+
+def config_parse_boolean(name: str, dest: str, value: Optional[str], namespace: argparse.Namespace) -> None:
+    setattr(namespace, dest, parse_boolean(value) if value is not None else False)
+
+
+def config_match_boolean(name: str, dest: str, value: str, namespace: argparse.Namespace) -> bool:
+    return bool(getattr(namespace, dest) == parse_boolean(value))
+
+
+def config_parse_feature(name: str, dest: str, value: Optional[str], namespace: argparse.Namespace) -> None:
+    if value is None:
+        value = "auto"
+    setattr(namespace, dest, parse_boolean(value) if value != "auto" else None)
+
+
+def config_parse_compression(name: str, dest: str, value: Optional[str], namespace: argparse.Namespace) -> None:
+    if value in ("zlib", "lzo", "zstd", "lz4", "xz"):
+        setattr(namespace, dest, value)
+    else:
+        setattr(namespace, dest, parse_boolean(value) if value is not None else None)
+
+
+def config_parse_base_packages(name: str, dest: str, value: Optional[str], namespace: argparse.Namespace) -> None:
+    if value == "conditional":
+        setattr(namespace, dest, value)
+    else:
+        setattr(namespace, dest, parse_boolean(value) if value is not None else False)
+
+
+def config_parse_distribution(name: str, dest: str, value: Optional[str], namespace: argparse.Namespace) -> None:
+    assert value is not None
+
+    try:
+        d = Distribution[value]
+    except KeyError:
+        die(f"Invalid distribution {value}")
+
+    r = {
+        Distribution.fedora: "37",
+        Distribution.centos: "9",
+        Distribution.rocky: "9",
+        Distribution.alma: "9",
+        Distribution.mageia: "7",
+        Distribution.debian: "testing",
+        Distribution.ubuntu: "jammy",
+        Distribution.opensuse: "tumbleweed",
+        Distribution.openmandriva: "cooker",
+        Distribution.gentoo: "17.1",
+    }.get(d, "rolling")
+
+    setattr(namespace, dest, d)
+    setattr(namespace, "release", r)
+
+
+ConfigParseCallback = Callable[[str, str, Optional[str], argparse.Namespace], None]
+ConfigMatchCallback = Callable[[str, str, str, argparse.Namespace], bool]
+
+@dataclasses.dataclass(frozen=True)
+class ConfigSetting:
+    dest: str
+    section: str
+    parse: ConfigParseCallback = config_parse_string
+    match: Optional[ConfigMatchCallback] = None
+    name: str = ""
+    default: Any = None
+    paths: tuple[str, ...] = tuple()
+
+    def __post_init__(self) -> None:
+        if not self.name:
+            object.__setattr__(self, 'name', ''.join(x.capitalize() for x in self.dest.split('_') if x))
+
+
+class ConfigParser:
```

Comment:
Can we name this a bit differently so as to make it clearly different from `configparser.ConfigParser`? Maybe just
```suggestion
class MkosiConfigParser:
```